### ISSUE TITLE
Version 2.6.2: Custom Category/Style Mismatch

### ISSUE THREAD

**Issue by Simon-Haynes** [geoffhumphrey/brewcompetitiononlineentry#1528]
**Verify the Following**
Reporting About the Latest Version? 2.6.2
*If you are not running the most current version in your environment, please update prior to reporting your bug. Your issue may have already been addressed.*

Searched Issue Reports for Bug? Yes
*Please search open *and* closed issues for issues similar to yours. Your bug may already have a posted solution.*

Reviewed the Good To Know Issue Reports? Yes
*Please review the Issue reports that are labeled Good to Know. Find one and click on the label to view them all.*

**BCOE&M Info**
Version: 2.6.2
Installation URL: https://pro-am24.nswhomebrewing.org/
Hosted Installation? No

**Describe the Bug**
I have added a number of custom styles (a lot) and most appear to be OK, but for some there is a Custom Category/Style mismatch i.e. when adding a custom style a custom category is automatically created for it on a 1:1 basis. For some reason a couple of custom styles have been associated with the custom category for another custom style (see attached) I am happy to edit the database directly if you can point me to the appropriate database file.

**To Reproduce**
1. Go to 'Styles Accepted'
2. Click on 'Add...'
3. Scroll down to 'A Custom Style'
4. See error 'Some custom styles associated with the custom category for another custom style.

**Expected Behavior**
Each custom style should be associated with its own custom category not that from another custom style.

**Screenshots**
If applicable, add screenshots to help explain your problem (see attached).

**Server Environment**
- PHP Version: 8.1.28
- MySQL Version: mysqlnd 8.1.28
- Other Info: cpsrvd 11.120.0.5

**Local Environment**
 - OS: Windows 10
 - Browser: Chrome
 - Browser Version: 124.0.6367.208
![CategoryStyleConflict](https://github.com/geoffhumphrey/brewcompetitiononlineentry/assets/126308048/b62c424f-a322-4b1b-bade-45b09486d9b1)

**Additional Context**

**Comment by Simon-Haynes** [comment]
I have realised, I think correctly, the the first three characters of a custom style have to be unique to distinguish it from other custom styles, which sort of makes the second two characters somewhat redundant. I guess it could be used to add a custom ‘category’ such as ‘Cold Brew’ e.g CLD00 then disable that style and add ‘Cold Brew Pale’ CLD01 and ‘Cold Brew Dark’ CLD02. Another quirk that I found was adding a custom style with a leading zero appears to be successful, but then when adding an entry for that style the description isn’t shown in the list of entries.

**Comment by geoffhumphrey** [comment]
The characters of the _Style Number or Identifier_ does not have to be unique. However, the combination of the _Style Number or Identifier_ AND the _Sub-Style Number or Identifier_ MUST be unique.

To alleviate the issue you reported, for the next version, I've added more robust back-end/real-time checking of the Style/Sub-Style combinations to make extra sure that there cannot be any duplicates. If a duplicate combination is found, the user will not be able to save the custom style (the Add Custom Style button will be disabled).

For example, I want to add a custom style called Black IPA. The combination of IPA as a style identifier and BL for a sub-style identifier is OK:
![image](https://github.com/user-attachments/assets/4a2bc9a5-1d17-4576-a9c1-05fd9d98fe36)

However, if I want to add a new style called Double Black IPA, I can't use that same combination.
![image](https://github.com/user-attachments/assets/bad921a8-ff0b-4a5f-8233-104ea6ff4ea4)

Changing the sub-style identifier to DB will allow the new style to be created.
![image](https://github.com/user-attachments/assets/ea13e0ea-e413-4d1b-856e-8032f296d274)

![image](https://github.com/user-attachments/assets/b25a206a-55cd-4e8f-b77f-7028fb7d72f8)

**Comment by Simon-Haynes** [comment]
Hi Geoff,

Apologies for contacting you directly, but we have run into a major issue
with displaying our results.

https://pro-am24.nswhomebrewing.org/

We changed the ‘display the results’ date/time to show the results
immediately and then realised we needed to change the placings for two
tables. So we pushed the display date back, changed the placings and then
brought the date/time forward again, but now, none of the table placings
are showing?

All the scores and placings are in the system and the best club results are
showing, but not the table/medal tables? The tables themselves are showing
no entries received and it’s no longer possible to see the entries and
scores for each table. Any suggestions?

Kind regards,

Simon Haynes.
